app/controllers/beaches_controller.rb

```diff
@@ -65,6 +65,42 @@ def destroy
 
   private
 
+  def search(query)
+    if query == nil
+      @beaches = Beach.all
+    elsif query.match(/^\s*$/)
+      flash[:notice] = "Please enter one or more words to search."
+      @beaches = Beach.all
+    elsif query.split.size == 1
+      @beaches = Beach.joins(:amenities).where(
+        "beaches.name ILIKE ? OR
+        beaches.description ILIKE ? OR
+        beaches.city ILIKE ? OR
+        beaches.state ILIKE ? OR
+        beaches.zip ILIKE ? OR
+        amenities.name ILIKE ?",
+        "%#{query}%", "%#{query}%", "%#{query}%", "%#{query}%", "%#{query}%",
+        "%#{query}%"
+      )
+    elsif query.split.size >= 2
+      container = []
+      split_query = query.split
+      mapped_split_query = split_query.map do |word|
```

Comment:
Unused block argument - `word`. You can omit the argument if you don't care about it.